pkg/apis/extensions/types.go

```diff
@@ -228,6 +228,14 @@ type DeploymentSpec struct {
 	// Value of this key is hash of DeploymentSpec.PodTemplateSpec.
 	// No label is added if this is set to empty string.
 	UniqueLabelKey string `json:"uniqueLabelKey,omitempty"`
+
+	// The config this deployment is rolling back to. Will be cleared after rollback is done.
+	Rollback *RollbackConfig `json:"rollback,omitempty"`
+}
+
+type RollbackConfig struct {
```

Comment:
Note: I do realize that the Annotations field is different than what we did for Binding. I think the simplicity gained is worth the inconsistency.